distribution/pull_v2.go

```diff
@@ -245,6 +245,8 @@ func (ld *v2LayerDescriptor) Download(ctx context.Context, progressOutput progre
 				return nil, 0, xfer.DoNotRetry{Err: err}
 			}
 			return nil, 0, err
+		} else if err == distribution.ErrBlobUnknown {
```

Comment:
Updated to move to `retryOnError`, and removed the other special casing of ErrBlobUnknown to follow the same pattern.